src/owner.rs

```diff
@@ -0,0 +1,108 @@
+use std::{mem, slice};
+
+use ndarray::{ArrayBase, Dimension, OwnedRepr};
+use pyo3::class::impl_::{PyClassImpl, ThreadCheckerStub};
+use pyo3::pyclass::PyClass;
+use pyo3::pyclass_slots::PyClassDummySlot;
+use pyo3::type_object::{LazyStaticType, PyTypeInfo};
+use pyo3::{ffi, types::PyAny, PyCell};
+
+use crate::dtype::Element;
+
+pub(crate) struct Owner {
+    ptr: *mut u8,
+    len: usize,
+    cap: usize,
+    drop: unsafe fn(*mut u8, usize, usize),
+}
+
+unsafe impl Send for Owner {}
+
+impl<T: Send> From<Box<[T]>> for Owner {
+    fn from(data: Box<[T]>) -> Self {
+        unsafe fn drop_boxed_slice<T>(ptr: *mut u8, len: usize, _cap: usize) {
+            let _ = Box::from_raw(slice::from_raw_parts_mut(ptr as *mut T, len) as *mut [T]);
+        }
+
+        let ptr = data.as_ptr() as *mut u8;
+        let len = data.len();
+        let cap = 0;
+        let drop = drop_boxed_slice::<T>;
+
+        mem::forget(data);
+
+        Self {
+            ptr,
+            len,
+            cap,
+            drop,
+        }
+    }
+}
+
+impl<T: Send> From<Vec<T>> for Owner {
+    fn from(data: Vec<T>) -> Self {
+        unsafe fn drop_vec<T>(ptr: *mut u8, len: usize, cap: usize) {
+            let _ = Vec::from_raw_parts(ptr as *mut T, len, cap);
+        }
+
+        let ptr = data.as_ptr() as *mut u8;
+        let len = data.len();
+        let cap = data.capacity();
+        let drop = drop_vec::<T>;
+
+        mem::forget(data);
```

Comment:
I prefer `Vec::into_raw_parts` as above, but it's a still nightly API.
So, leaving a TODO comment would be nice, like:
```suggestion
         // TODO: Use Vec::into_raw_parts when it's stabilized
        mem::forget(data);
```

Reply:
Committed as suggested.